bolt-sidecar/src/state/consensus.rs

```diff
@@ -118,3 +133,50 @@ fn current_timestamp() -> u64 {
         .unwrap()
         .as_secs()
 }
+
+/// Filters the proposer duties and returns the validator index for a given slot
+/// if it doesn't exists then returns 0 by default.
+pub fn find_validator_index_for_slot(
```

Comment:
This should be a method on `ConsensusState`, which already has `validator_indexes` and `proposer_duties` as fields

Reply:
good catch, done